spec/requests/common_area_request_spec.rb

```diff
@@ -1,25 +1,69 @@
 require 'rails_helper'
 
-describe 'POST /common_areas' do
-  it 'must be authenticated to register a common area' do
-    condo = create(:condo)
+describe 'Common Areas' do
+  context 'GET /common_areas' do
+    it 'must be authenticated as Super Manager or condo associated' do
+      condo_manager = create :manager, is_super: false
+      condo = create :condo
+      common_area = create(:common_area, condo:)
 
-    post condo_common_areas_path(condo), params: { common_area: { name: 'Salão de Festas', max_occupancy: 100,
-                                                                  description: 'Realize sua festa em nosso salão',
-                                                                  rules: 'Não pode ser utilizado após as 22 horas' } }
+      login_as condo_manager, scope: :manager
+      get common_area_path common_area
 
-    expect(response).to redirect_to(new_manager_session_path)
-    expect(flash[:alert]).to eq 'Para continuar, faça login ou registre-se.'
+      expect(response).to redirect_to root_path
+    end
   end
 
-  it 'must be authenticated to edit a common area' do
-    common_area = create(:common_area)
+  context 'POST /common_areas' do
+    it 'must be authenticated to register a common area' do
+      condo = create :condo
 
-    patch common_area_path(common_area), params: { common_area: { name: 'Churrasqueira', max_occupancy: 30,
-                                                                  description: 'Reuna a família em um churrasco',
-                                                                  rules: 'Deixe o espaço organizado após o uso' } }
+      post condo_common_areas_path condo, params: { common_area: { name: 'Salão de Festas', max_occupancy: 100,
+                                                                   description: 'Realize sua festa em nosso salão',
+                                                                   rules: 'Não pode ser utilizado após as 22 horas' } }
 
-    expect(response).to redirect_to(new_manager_session_path)
-    expect(flash[:alert]).to eq 'Para continuar, faça login ou registre-se.'
+      expect(response).to redirect_to new_manager_session_path
+      expect(flash[:alert]).to eq 'Para continuar, faça login ou registre-se.'
+    end
+
+    it 'and he´s not associated or is not a super' do
```

Comment:
Esse `he´s` era para ser um apóstrofo, mas está uma crase.

Reply:
acho que não funciona deixar apóstrofo, senão o rails identifica como o fim de uma string, o ideal seria não usar a contração e sim "and he is not"